tests/benchmark/compute/instruction/test_system.py

```diff
@@ -108,6 +113,128 @@ def test_xcall(
             "during the setup phase of this test."
         )
 
+    initcode, factory_address, factory_caller_address = (
+        _deploy_max_contract_factory(pre, fork)
+    )
+
+    # Deploy num_contracts via multiple txs (each capped by tx gas limit).
+    with TestPhaseManager.setup():
+        # Rough estimate (rounded down) of contracts per tx based on dominant
+        # cost factor only. E.g., 17M gas limit + 24KiB contracts = ~3 per tx.
+        # The goal is to involve the minimum amount of gas pricing to avoid
+        # complexity and potential brittleness.
+        # If this estimation is incorrect in the future (i.e. tx gas limit cap)
+        # is increased or cost per byte, the post-state check will detect it
+        # and can be adjusted with a more complex formula.
+        num_contracts_per_tx = tx_gas_limit_cap // (
+            gas_costs.G_CODE_DEPOSIT_BYTE * max_contract_size
+        )
+        attack_txs = math.ceil(num_contracts / num_contracts_per_tx)
+
+        contracts_deployment_txs = []
+        for _ in range(attack_txs):
+            contracts_deployment_txs.append(
+                Transaction(
+                    to=factory_caller_address,
+                    gas_limit=tx_gas_limit_cap,
+                    gas_price=10**6,
+                    data=Hash(num_contracts_per_tx),
+                    sender=pre.fund_eoa(),
+                )
+            )
+
+    post = {}
+    for i in range(num_contracts):
+        deployed_contract_address = compute_create2_address(
+            address=factory_address,
+            salt=i,
+            initcode=initcode,
+        )
+        post[deployed_contract_address] = Account(nonce=1)
+
+    attack_call = Bytecode()
+    if opcode == Op.EXTCODECOPY:
+        attack_call = Op.EXTCODECOPY(
+            address=Op.SHA3(32 - 20 - 1, 85), dest_offset=96, size=1000
+        )
+    else:
+        # For the rest of the opcodes, we can use the same generic attack call
+        # since all only minimally need the `address` of the target.
+        attack_call = Op.POP(opcode(address=Op.SHA3(32 - 20 - 1, 85)))
+    attack_code = (
+        # Setup memory for later CREATE2 address generation loop.
+        # 0xFF+[Address(20bytes)]+[seed(32bytes)]+[initcode keccak(32bytes)]
+        Op.MSTORE(0, factory_address)
+        + Op.MSTORE8(32 - 20 - 1, 0xFF)
+        + Op.MSTORE(32, Op.CALLDATALOAD(0))
+        + Op.MSTORE(64, initcode.keccak256())
+        # Main loop
+        + While(
+            body=attack_call + Op.MSTORE(32, Op.ADD(Op.MLOAD(32), 1)),
+        )
+    )
+
+    if len(attack_code) > max_contract_size:
+        # TODO: A workaround could be to split the opcode code into multiple
+        # contracts and call them in sequence.
+        raise ValueError(
+            f"Code size {len(attack_code)} exceeds maximum "
+            f"code size {max_contract_size}"
+        )
```

Comment:
```suggestion
        # TODO: A workaround could be to split the opcode code into multiple
        # contracts and call them in sequence.
```

This check could be removed (we could leave the `TODO` as it is now). Since in both fill & execute mode, there is contract size limit check:

**Fill**
https://github.com/ethereum/execution-specs/blob/9523e75061bc9f989f0867e34f28364c719a3b07/packages/testing/src/execution_testing/cli/pytest_commands/plugins/filler/pre_alloc.py#L188-L191

**Execute**
https://github.com/ethereum/execution-specs/blob/9523e75061bc9f989f0867e34f28364c719a3b07/packages/testing/src/execution_testing/cli/pytest_commands/plugins/execute/pre_alloc.py#L325-L327

Reply:
Sounds good. I think I'll remove the comment since leaving it alone feels pretty weird without context.